src/java/main/org/apache/zookeeper/server/command/FourLetterCommands.java

```diff
@@ -153,13 +159,50 @@
      */
     public final static int telnetCloseCmd = 0xfff4fffd;
 
-    final static HashMap<Integer, String> cmd2String =
-        new HashMap<Integer, String>();
+    private static final String ZOOKEEPER_4LW_COMMANDS_WHITELIST = "zookeeper.4lw.commands.whitelist";
+
+    // A property only used in tests to turn on / off entire set of supported four letter word commands.
+    private static final String ZOOKEEPER_4LW_TEST = "zookeeper.test.4lw.enabled";
+
+    private static final Logger LOG = LoggerFactory.getLogger(FourLetterCommands.class);
+
+    private static final Map<Integer, String> cmd2String = new HashMap<Integer, String>();
+
+    private static final Set<String> whiteListedCommands = new HashSet<String>();
 
     public static Map<Integer, String> getCmdMapView() {
         return Collections.unmodifiableMap(cmd2String);
     }
 
+    // ZOOKEEPER-2693: Only allow white listed commands.
+    public static Set<String> getWhiteListedCmdView() {
```

Comment:
I think instead of returning all the commands all the time and making collection object. We can write function isWhiteListedCommand(String command) and use it